packages/mui-utils/src/usePreviousProps.ts

```diff
@@ -1,11 +1,15 @@
 import * as React from 'react';
 
-const usePreviousProps = (value: object): object => {
-  const ref = React.useRef({});
+type AddUndefined<T> = {
+  [Property in keyof T]: T[Property] | undefined;
```

Comment:
```suggestion
  [Property in keyof T]?: T[Property];
```

Strictly speaking the property is totally absent from the props when on the 1st render, not equal to undefined :+1: 
Even it is almost always the same thing it the end

Reply:
Ah yes there is subtle difference b/w both. i've updated code as suggested. Also thanks for the review